Server.Core/Reply.cs

```diff
@@ -34,6 +31,57 @@ public byte[] ReplyMessage()
             return wholeMessage;
         }
 
+        private byte[] GetBodyMessage()
+        {
+            if (!(_response.Body == null))
+            {
+                _response.ReadyStreamForRead();
+                var buffer = new byte[(int)_response.Body.Length];
+                _bodySize = _response.Body.Read(buffer, 0, buffer.Length);
+                return buffer;
+            }
+            return new byte[0];
+        }
+
+        private byte[] CreateHeaders()
+        {
+            if (_bodySize != 0)
+            {
+                if (!(_response.ContentType == null))
+                {
+                    return Encoding.UTF8.GetBytes("Content-Type: " + _response.ContentType + "\r\n" + "Content-Length: " + _bodySize + "\r\n");
```

Comment:
for this and the other methods that just have static text being encoded into bytes, it'd probably be less work to just deal with strings and then turn everything into bytes right before returning to the client